### DISCUSSION TITLE
Select referenced table in different schema

### DISCUSSION THREAD

**Discussion by iberis2** [supabase/supabase#28474] [Questions] [answered]
Hi, I was trying to select referenced table in other schema.

`user`, `sellers`, `likes` table is in clients schema and `products` table is in ecommerce schema.

when I select referenced table in different schema, it throw error.
I tried as below.

```ts
// try 1
const { data, error } =   await supabase
      .schema("clients")
      .from("users")
      .select(
        `*, sellers( likes( ecommerce.products(*)))`
      )
      
// try 2
const { data, error } =   await supabase
      .schema("clients")
      .from("users")
      .select(
        `*, sellers( likes(products(*)))`
      )
      

```

```json

// error 1
{
    "code": "PGRST100",
    "details": "unexpected \"p\" expecting \"sum\", \"avg\", \"count\", \"max\" or \"min\"",
    "hint": null,
    "message": "\"failed to parse select parameter (*,sellers(likes(ecommerce.products(*))))\" (line 1, column 70)"
}


// error 2

{
    "code": "PGRST200",
    "details": "Searched for a foreign key relationship between 'likes' and 'products' in the schema 'clients', but no matches were found.",
    "hint": null,
    "message": "Could not find a relationship between 'likes' and 'products' in the schema cache"
}
```

-----
https://github.com/orgs/supabase/discussions/6550

I found this discussion, but I can't believe that you all work with only one schema.
Is it true that Supabase still does not support multiple schemas?
Can you share how you work when different schemas reference each other?

**Comment by GaryAustin1** [comment, marked as answer]
The rest API only works on one schema at a time.  It is in an http header which one will be used.

You can use an rpc call to a Postgres function or a view to handle multiple schemas would be other approaches.

**Reply by iberis2** [reply]
Tanks for replying
